drivers/interrupt_controller/intc_vim.c

```diff
@@ -0,0 +1,150 @@
+/* Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/arch/arm/aarch32/irq.h>
+#include <zephyr/sys/util_macro.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(vim);
+
+#include <zephyr/drivers/interrupt_controller/intc_vim.h>
+
+unsigned int z_vim_irq_get_active(void)
+{
+	uint32_t irq_group_num, irq_bit_num;
+	uint32_t actirq, vec_addr;
+
+	/* Reading IRQVEC register, ACTIRQ gets loaded with valid IRQ values */
+	vec_addr = sys_read32(VIM_IRQVEC);
+
+	/* ACTIRQ register should be read only after reading IRQVEC register */
+	actirq = sys_read32(VIM_ACTIRQ);
+
+	/* Check if the irq number is valid, else return invalid irq number.
+	 * which will be considered as spurious interrupt
+	 */
+	if ((actirq & (VIM_ACTIRQ_VALID_MASK)) == 0) {
+		return CONFIG_NUM_IRQS + 1;
+	}
+
+	irq_group_num = VIM_GET_IRQ_GROUP_NUM(actirq & VIM_PRIIRQ_NUM_MASK);
+	irq_bit_num = VIM_GET_IRQ_BIT_NUM(actirq & VIM_PRIIRQ_NUM_MASK);
+
+	/* Ack the interrupt in IRQSTS register */
+	sys_write32(BIT(irq_bit_num), VIM_IRQSTS(irq_group_num));
+
+	if (irq_group_num > VIM_MAX_GROUP_NUM) {
+		return (CONFIG_NUM_IRQS + 1);
+	}
+
+	return (actirq & VIM_ACTIRQ_NUM_MASK);
+}
+
+void z_vim_irq_eoi(unsigned int irq)
+{
+	sys_write32(0, VIM_IRQVEC);
+}
+
+void z_vim_irq_init(void)
+{
+	uint32_t num_of_irqs = sys_read32(VIM_INFO_INTERRUPTS_MASK);
+
+	LOG_PRINTK("VIM: Number of IRQs = %u\n", num_of_irqs);
```

Comment:
why `LOG_PRINTK` and not `LOG_DBG` or something like that?

Reply:
Changing to LOG_DBG.